app/components/post_draft/draft_input/draft_input.test.tsx

```diff
@@ -0,0 +1,166 @@
+        channelId: 'channelId',
+        channelType: 'O' as ChannelType,
+        currentUserId: 'currentUserId',
+        postPriority: {priority: ''} as PostPriority,
+        updatePostPriority: jest.fn(),
+        persistentNotificationInterval: 0,
+        persistentNotificationMaxRecipients: 0,
+        updateCursorPosition: jest.fn(),
+        cursorPosition: 0,
+        sendMessage: jest.fn(),
+        canSend: true,
+        maxMessageLength: 4000,
+        files: [],
+        value: '',
+        uploadFileError: null,
+        updateValue: jest.fn(),
+        addFiles: jest.fn(),
+        updatePostInputTop: jest.fn(),
+        setIsFocused: jest.fn(),
+        scheduledPostsEnabled: true,
+    };
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    let database: Database;
+    let operator: ServerDataOperator;
+
+    beforeEach(async () => {
+        const server = await TestHelper.setupServerDatabase(SERVER_URL);
+        database = server.database;
+        operator = server.operator;
+    });
+
+    afterEach(async () => {
+        await TestHelper.tearDown();
+        NetworkManager.invalidateClient(SERVER_URL);
+    });
+
+    describe('Rendering', () => {
+        it('renders base components', async () => {
+            const {getByTestId} = renderWithEverything(<DraftInput {...baseProps}/>, {database});
+            expect(getByTestId('draft_input')).toBeTruthy();
+            expect(getByTestId('draft_input.post.input')).toBeTruthy();
+            expect(getByTestId('draft_input.quick_actions')).toBeTruthy();
+            expect(getByTestId('draft_input.send_action.send.button')).toBeTruthy();
+        });
+
+        it('shows upload error when present', () => {
+            const props = {...baseProps, uploadFileError: 'Error message'};
+            const {getByText} = renderWithEverything(<DraftInput {...props}/>, {database});
+            expect(getByText('Error message')).toBeTruthy();
+        });
+    });
+
+    describe('Message Actions', () => {
+        it('sends message on press', () => {
+            const {getByTestId} = renderWithEverything(<DraftInput {...baseProps}/>, {database});
+            fireEvent.press(getByTestId('draft_input.send_action.send.button'));
+            expect(baseProps.sendMessage).toHaveBeenCalled();
+        });
+
+        it('opens scheduled post options on long press', async () => {
+            // make this a re-usable function
+            await operator.handleConfigs({
+                configs: [
+                    {id: 'ScheduledPosts', value: 'true'},
+                ],
+                configsToDelete: [],
+                prepareRecordsOnly: false,
+            });
+
+            const {getByTestId} = renderWithEverything(<DraftInput {...baseProps}/>, {database});
+            fireEvent(getByTestId('draft_input.send_action.send.button'), 'longPress');
+            expect(openAsBottomSheet).toHaveBeenCalledWith(expect.objectContaining({
```

Comment:
Just as sanity check, we could add here also a check to make sure the other function (`sendMessage` if I am not mistaken) is not called.

Reply:
Done.